src/components/dashboard/main-content/course-enrollments/data/hooks.js

```diff
@@ -525,29 +527,67 @@ export function useCourseEnrollmentsBySection(courseEnrollmentsByStatus) {
+  queryClient,
+  enterpriseSlug,
+  enterpriseCustomer,
+  courseRunId,
+  updatedEnrollmentParams,
+}) {
+  // Transformation
+  const { newStatus } = updatedEnrollmentParams;
+  const transformUpdatedEnrollment = (enrollment) => {
+    if (enrollment.courseRunId !== courseRunId) {
+      return enrollment;
+    }
+    return {
+      ...enrollment,
+      courseRunStatus: newStatus,
+    };
+  };
+  const isBFFEnabled = isBFFEnabledForEnterpriseCustomer(enterpriseCustomer.uuid);
+
+  if (isBFFEnabled) {
+  // Determine which BFF queries need to be updated after unenrolling.
+    const dashboardBFFQueryKey = queryEnterpriseLearnerDashboardBFF(
+      { enterpriseSlug },
+    ).queryKey;
+    const bffQueryKeysToUpdate = [dashboardBFFQueryKey];
+    // Update the enterpriseCourseEnrollments data in the cache for each BFF query.
+    bffQueryKeysToUpdate.forEach((queryKey) => {
+      const existingBFFData = queryClient.getQueryData(queryKey);
+      const updatedBFFData = {
+        ...existingBFFData,
+        enterpriseCourseEnrollments: existingBFFData.enterpriseCourseEnrollments.map(transformUpdatedEnrollment),
+      };
+      queryClient.setQueryData(queryKey, updatedBFFData);
+    });
+  } else {
+    // Update the legacy queryEnterpriseCourseEnrollments cache as well.
+    const enterpriseCourseEnrollmentsQueryKey = queryEnterpriseCourseEnrollments(enterpriseCustomer.uuid).queryKey;
+    const existingCourseEnrollmentsData = queryClient.getQueryData(enterpriseCourseEnrollmentsQueryKey);
+    const updatedCourseEnrollmentsData = existingCourseEnrollmentsData.map(transformUpdatedEnrollment);
+    queryClient.setQueryData(enterpriseCourseEnrollmentsQueryKey, updatedCourseEnrollmentsData);
+  }
+}
+
 export const useUpdateCourseEnrollmentStatus = ({ enterpriseCustomer }) => {
   const queryClient = useQueryClient();
+  const params = useParams();
+  const location = useLocation();
 
-  const updateCourseEnrollmentStatus = useCallback(({ courseRunId, newStatus, savedForLater }) => {
-    const enrollmentsQueryKey = queryEnterpriseCourseEnrollments(enterpriseCustomer.uuid).queryKey;
-    const existingEnrollments = queryClient.getQueryData(enrollmentsQueryKey);
-    queryClient.setQueryData(
-      enrollmentsQueryKey,
-      existingEnrollments.map((enrollment) => {
-        if (enrollment.courseRunId === courseRunId) {
-          return {
-            ...enrollment,
-            courseRunStatus: newStatus,
-            savedForLater,
-          };
-        }
-        return enrollment;
-      }),
-    );
-  }, [
-    enterpriseCustomer.uuid,
-    queryClient,
-  ]);
+  const updateCourseEnrollmentStatus = useCallback(({ courseRunId, newStatus }) => {
+    handleQueriesForUpdatedCourseEnrollmentStatus({
+      queryClient,
+      location,
+      enterpriseSlug: params.enterpriseSlug,
+      enterpriseCustomer,
+      courseRunId,
+      updatedEnrollmentParams: {
+        newStatus,
+      },
```

Comment:
nit: If there's only 1 param that gets updated now (the enrollment status), I wonder if it's worth simplifying this arg to be `newEnrollmentStatus` vs. passing it via `updatedEnrollmentParams`.